### DISCUSSION TITLE
Failure to load any static files when deploying with HTTPS

### DISCUSSION THREAD

**Discussion by kareemamrr** [fastapi/fastapi#6950] [Questions] [answered]
### Code
```css
<head>
    <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700&display=swap" rel="stylesheet">
    <link href="{{ url_for('static', path='style.css') }}" rel="stylesheet">
</head
```

### Description
- I tested the code over HTTP and it was working but when deployed with NGINX over HTTPS it throws this error `Mixed Content: The page at __ was loaded over HTTPS, but requested an insecure stylesheet '__/static/style.css'. This request has been blocked; the content must be served over HTTPS.`
- There was another warning similar to this error regarding a svg file but it said it automatically "upgraded" the request to HTTPS.
- How can I fix this so that it can load my stylesheet and script file?

### Environment
- OS: Linux
- FastAPI version: 0.61.1
- Python version: 3.8.2

**Comment by falkben** [comment]
these things are always tricky. whatever you are using to host the app (uvicorn?) needs to be made aware of ssl. if this is in fact your issue, have a look at the uvicorn docs on this https://www.uvicorn.org/deployment/#running-with-https

**Comment by kareemamrr** [comment]
Hey @falkben, I actually just solved my previous issue by writing out the entire resource path but with `https://`. I then encountered another problem which is hitting endpoints of fastapi from the javascript part of the application. I am using NGINX infront of the fastapi application, which is deployed on the localhost using Uvicorn with Gunicorn as a process manager (`gunicorn -k uvicorn.workers.UvicornWorker`). When I try to hit an endpoint from javascript with
```js
 clear.onclick = e => {
    const XHR = new XMLHttpRequest();
    XHR.onreadystatechange = function () {
        if (XHR.readyState == XMLHttpRequest.DONE) {
             alert(XHR.responseText);
        }
     }
     XHR.open('GET', "127.0.0.1:8000/clear", true);
     XHR.send();
}
```
it returns `GET http://127.0.0.1:8000/clear net::ERR_CONNECTION_REFUSED`.

**Comment by falkben** [comment, marked as answer]
Have a look at x-forwarded-for and x-forwarded-proto headers. You should set proxy-headers in uvicorn.  You can see how the headers are used by uvicorn here: https://github.com/encode/uvicorn/blob/0.13.3/uvicorn/middleware/proxy_headers.py#L30-L43

This SO post helped me: https://stackoverflow.com/questions/63511413/fastapi-redirection-for-trailing-slash-returns-non-ssl-link

**Comment by kareemamrr** [comment]
Thanks!

**Comment by tiangolo** [comment]
Thanks for the help here @falkben ! 👏  🙇 

Thanks for reporting back and closing the issue @kareemamrr  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.

**Comment by nafets33** [comment]
hey I'm running into this issue using ngnix running through an https pollenq.com 

I'm serving using react-components through streamlit and having a difficult time implenting the solutions here and on SO https://stackoverflow.com/questions/63511413/fastapi-redirection-for-trailing-slash-returns-non-ssl-link

could i buy someone's time ?
